backend/moonstreamapi/routes/queries.py

```diff
@@ -0,0 +1,384 @@
+            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
+            entry_id=entry.id,
+            tags=[f"query_id:{entry.id}", f"preapprove"],
+        )
+
+    except BugoutResponseException as e:
+        logger.error(f"Error in applind tags to query entry: {str(e)}")
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+    except Exception as e:
+        logger.error(f"Error in applind tags to query entry: {str(e)}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    return entry
+
+
+@router.get("/{query_name}/query", tags=["queries"])
+async def get_query_handler(request: Request, query_name: str) -> BugoutJournalEntry:
+
+    token = request.state.token
+
+    query_id = get_query_by_name(query_name, token)
+
+    try:
+
+        entry = bc.get_entry(
+            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
+            entry_id=query_id,
+        )
+
+    except BugoutResponseException as e:
+        logger.error(f"Error in updating query: {str(e)}")
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+
+    except Exception as e:
+        logger.error(f"Error in updating query: {e}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    return entry
+
+
+@router.put("/{query_name}", tags=["queries"])
+async def update_query_handler(
+    request: Request,
+    query_name: str,
+    request_update: data.UpdateQueryRequest = Body(...),
+) -> BugoutJournalEntryContent:
+
+    token = request.state.token
+
+    query_id = get_query_by_name(query_name, token)
+
+    try:
+
+        entry = bc.update_entry_content(
+            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
+            entry_id=query_id,
+            title=query_name,
+            content=request_update.query,
+            tags=["preapprove"],
+        )
+
+    except BugoutResponseException as e:
+        logger.error(f"Error in updating query: {str(e)}")
+        raise MoonstreamHTTPException(status_code=e.status_code, detail=e.detail)
+
+    except Exception as e:
+        logger.error(f"Error in updating query: {e}")
+        raise MoonstreamHTTPException(status_code=500, internal_error=e)
+
+    return entry
+
+
+@router.post("/{query_name}/update_data", tags=["queries"])
+async def update_query_data_handler(
+    request: Request,
+    query_name: str,
+    request_update: data.UpdateDataRequest = Body(...),
+) -> Optional[Dict[str, Any]]:
+    """
+    Request update data on S3 bucket
+    """
+
+    token = request.state.token
+
+    query_id = get_query_by_name(query_name, token)
+
+    try:
+        entries = bc.search(
+            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
+            query=f"#approved #query_id:{query_id} !#preapprove",

```

Comment:
`!#preapprove` is not working correctly (I think better just to use it) and better to use here `tag:` instead of `#`